### PR TITLE
Replace AttributeConsumer with BiConsumer

### PR DESCRIPTION
**Pull request by anuraaga**
Our current interface has a wildcard type parameter - it seems to provide no real benefit in almost all cases, but prevents using lambdas with `Attributes.forEach`. For the cases it helped a bit, `putAll` works fine.

### CODE REVIEW

**Comment by anuraaga** [comment]
This is breaking, but I don't think I can do it with deprecation - if we had two `forEach` methods, the caller would need to pick with an explicit cast.

File: api/src/main/java/io/opentelemetry/api/common/ArrayBackedAttributesBuilder.java [hunk 1]
```diff
@@ -90,8 +90,11 @@ public AttributesBuilder put(String key, boolean... value) {
   }
 
   @Override
-  public AttributesBuilder putAll(Attributes attributes) {
-    attributes.forEach(this::put);
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public AttributesBuilder putAll(ReadableAttributes attributes) {
+    // Attributes must iterate over their entries with matching types for key / value, so this
+    // downcast to the raw type is safe.
+    attributes.forEach((key, value) -> put((AttributeKey) key, value));
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/api/common/ArrayBackedAttributesBuilder.java:97]: Populating another attributes is basically the only time `<T>` helped - but we can manage that ourselves without sacrificing the ability to lambda

**Comment by jkwatson** [comment]
Both options end up being kind-of ugly. At least this one has one fewer public thing we have to maintain. The lack of union types is definitely hurting us here, for sure.

**Comment by jkwatson** [comment]
@marcingrzejszczak another small breaking change here.

**Review by jkwatson — APPROVED** [review]
👍

**Pull request merged**